client/Windows/wf_floatbar.c

```diff
@@ -170,51 +181,114 @@ static Button* floatbar_create_lock_button(FloatBar* floatbar,
 
 	button->unlocked_bmp = button->bmp;
 	button->unlocked_bmp_act = button->bmp_act;
-	button->locked_bmp = (HBITMAP)LoadImage(floatbar->wfc->hInstance, MAKEINTRESOURCE(lock_resid), IMAGE_BITMAP, w, h, LR_DEFAULTCOLOR);
-	button->locked_bmp_act = (HBITMAP)LoadImage(floatbar->wfc->hInstance, MAKEINTRESOURCE(lock_resid_act), IMAGE_BITMAP, w, h, LR_DEFAULTCOLOR);
+	button->locked_bmp = (HBITMAP)LoadImage(floatbar->wfc->hInstance, MAKEINTRESOURCE(lock_resid), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+	button->locked_bmp_act = (HBITMAP)LoadImage(floatbar->wfc->hInstance, MAKEINTRESOURCE(lock_resid_act), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
 
 	return button;
 }
 
-static Button* floatbar_get_button(FloatBar* floatbar, int x, int y)
+static Button* floatbar_get_button( const FloatBar * const floatbar, const int x, const int y)
 {
 	int i;
 
 	if (y > BUTTON_Y && y < BUTTON_Y + BUTTON_HEIGHT)
+    {
 		for (i = 0; i < BTN_MAX; i++)
-			if (x > floatbar->buttons[i]->x && x < floatbar->buttons[i]->x + floatbar->buttons[i]->w)
+        {
+            if (floatbar->buttons[i] != NULL && x > floatbar->buttons[i]->x && x < floatbar->buttons[i]->x + floatbar->buttons[i]->w)
+            {
 				return floatbar->buttons[i];
+            }
+        }
+    }
 
 	return NULL;
 }
 
-static int floatbar_paint(FloatBar* floatbar, HDC hdc)
+static int floatbar_paint( FloatBar * const floatbar, const HDC hdc)
 {
 	int i;
 
 	/* paint background */
-	SelectObject(floatbar->hdcmem, floatbar->background);
-	StretchBlt(hdc, 0, 0, BACKGROUND_W, BACKGROUND_H, floatbar->hdcmem, 0, 0, BACKGROUND_W, BACKGROUND_H, SRCCOPY);
+  
+    GRADIENT_RECT gradientRect = { 0, 1 };
+    COLORREF rgbTop = RGB( 117, 154, 198 );
+    COLORREF rgbBottom = RGB( 6, 55, 120 );
+
+    const int top = 0;
+    int left = 0;
+    int bottom = BACKGROUND_H - 1;
+    int right = BACKGROUND_W - 1;
+    const int angleOffset = BACKGROUND_H - 1;
+
+    TRIVERTEX triVertext[2] = {
+        left,
+        top,
+        GetRValue( rgbTop ) << 8,
+        GetGValue( rgbTop ) << 8,
+        GetBValue( rgbTop ) << 8,
+        0x0000,
+        right,
+        bottom,
+        GetRValue( rgbBottom ) << 8,
+        GetGValue( rgbBottom ) << 8,
+        GetBValue( rgbBottom ) << 8,
+        0x0000
+    };
+
+    GradientFill( hdc, triVertext, 2, &gradientRect, 1, GRADIENT_FILL_RECT_V );
+
+    /* paint shadow */
+
+    HPEN hpen = CreatePen( PS_SOLID, 1, RGB( 71, 71, 71 ) );
```

Comment:
Yes, and generally I like to check for errors but what should happen in the unlikely event GradientFill() or CreatePen() failed?  Most likely the entire graphics subsystem would be hosed and nothing would work if that were the case.  I thought about adding logging messages, but I couldn't locate any documentation on the logging system and don't have time to try and figure it out from the code (deadline at work, unfortunately).